### ISSUE TITLE
Problem with link rendering

### ISSUE THREAD

**Issue by JokerQyou** [russross/blackfriday#223]
I'm not an end user of blackfriday, but one of the developers of [gogs][1] [suggested me to open an issue here as well][2]. 
So, the issue is:

The following content will not be rendered correctly.
```markdown
This is a sentence with [a link][1] and [another][2].

  [1]: http://a.com
  [2]: http://b.com
```

Instead, a space is needed for correctly rendering of the second link, like this:

```markdown
This is a sentence with [a link][1] and [another][2].

  [1]: http://a.com_
  [2]: http://b.com_
```

Notice there is a space after each of the two links, which I replaced with underline symbol.

The former content will produce: the first link being rendered correctly, but the second one will be rendered as is; the second link will remain the form of:
```markdown
... [another][2].

  [2]: http://b.com
```

I don't think the tailing space is required in markdown spec. Correct me if I was wrong.

  [1]: https://github.com/gogits/gogs
  [2]: https://github.com/gogits/gogs/issues/2058#issuecomment-162077228

**Comment by twhitacre** [comment]
I can confirm this as well. I started using [Hugo](https://github.com/spf13/hugo) the other night and am running into the same issue.

I'll also point out that you can have as many reference links as you want, but if there is not space or new line after the last one listed, then the last one will not render. 

For instance:

```
This is a sentence with [a link][1] and [another][2] and [another][3] and [another][4].

  [1]: http://a.com
  [2]: http://b.com
  [3]: http://c.com
  [4]: http://d.com
```

all of those will work except for link `[4]`. If however you add a space or a return _after_ link `[4]` then it will render properly.

**Comment by twhitacre** [comment]
Also, I bet [this issue](https://github.com/russross/blackfriday/issues/180) has to do with the same problem.

**Comment by JokerQyou** [comment]
Hmm, I do know it's a tradition to have a newline at the end of file, but I don't see why this is necessary, especially when we're posting text content on the web.

**Comment by shurcooL** [comment]
> I don't think the tailing space is required in markdown spec. Correct me if I was wrong.

I agree. I think this is a bug/enhancement that should be implemented.

**Comment by rtfb** [comment]
Yeah, this is certainly a duplicate of #180. I'm closing this one and will push a fix for #180 in a few minutes.
